### ISSUE TITLE
unnecessary immutability

### ISSUE THREAD

**Issue by yegor256** [yegor256/cactoos#180]
I believe we made a mistake in:

```
ArrayAsIterable
IterableAsList
IterableAsMap
MapAsProperties
```

They must not cache what they calculate. Instead, we have to introduce `StickyList` and `StickyMap`, which will do what we need to prevent multiple recalculations.

**Comment by yegor256** [comment]
I'll fix it now

**Comment by yegor256** [comment]
fixed in #186
